configuration/configuration.go

```diff
@@ -244,6 +244,16 @@ func DefaultConfiguration() *Configuration {
 	}
 }
 
+// EndCondition contains all the conditions for the syncer to stop.
+type EndCondition struct {
```

Comment:
Should we move the "--end" flag here? I'm very tempted. (Set to EndIndex)

Reply:
I was tempted as well. I'd like to make another PR to address this